usd_qtpy/render_util/dialog.py

```diff
@@ -0,0 +1,889 @@
+            self.btn_browse_turntable.setDisabled(False)
+            self.lbl_turntable_file.setDisabled(False)
+        else:
+            self.txt_turntable_filename.setDisabled(True)
+            self.btn_browse_turntable.setDisabled(True)
+            self.lbl_turntable_file.setDisabled(True)
+
+    def frame_range_callback(self, formlayout: QtWidgets.QFormLayout):
+        """
+        Frame range is a different story for turntables.
+        It's defined by a starting frame, and a length
+        """
+        
+        self.spinbox_frame_start = QtWidgets.QSpinBox()
+        self.spinbox_frame_length = QtWidgets.QSpinBox()
+
+        self.spinbox_frame_start.setMinimum(-9999)
+        self.spinbox_frame_start.setMaximum(9999)
+        self.spinbox_frame_length.setMinimum(1)
+        self.spinbox_frame_length.setMaximum(9999)
+
+        self.spinbox_frame_start.setValue(1)
+        self.spinbox_frame_length.setValue(100)
+
+        framerange_hlayout = QtWidgets.QHBoxLayout()
+        framerange_hlayout.addWidget(self.spinbox_frame_start)
+        framerange_hlayout.addWidget(self.spinbox_frame_length)
+
+        formlayout.addRow("Starting Frame / Length", framerange_hlayout)
+
+        self.spinbox_loops = QtWidgets.QSpinBox()
+        self.spinbox_loops.setMinimum(1)
+        self.spinbox_loops.setMaximum(99)
+        formlayout.addRow("Repetitions", self.spinbox_loops)
+
+    def ui_pre_hook(self, vlayout: QtWidgets.QVBoxLayout):
+                
+        pre_form = QtWidgets.QFormLayout()
+        pre_form.setHorizontalSpacing(150)
+
+        # Turntable type chooser
+        self.cbox_turntable_type = QtWidgets.QComboBox()
+        turntable_types = ("Rotate camera",
+                           "Rotate subject",
+                           "Preset from file")
+        self.cbox_turntable_type.addItems(turntable_types)
+        pre_form.addRow("Turntable Type", self.cbox_turntable_type)
+        
+        # Turntable file chooser
+        self.lbl_turntable_file = QtWidgets.QLabel(self)
+        self.lbl_turntable_file.setText("Path to get turntable from...")
+        self.lbl_turntable_file.setFixedHeight(30)
+        pre_form.addRow(self.lbl_turntable_file)
+
+        self.txt_turntable_filename = QtWidgets.QLineEdit()
+        self.txt_turntable_filename.setPlaceholderText("Empty - Use internal preset")
+        self.btn_browse_turntable = QtWidgets.QPushButton(icon=get_icon("folder"))
+        self.btn_browse_turntable.setFixedSize(QtCore.QSize(30, 30))
+        self.btn_browse_turntable.clicked.connect(self._prompt_turntablefile)
+
+        turntable_filename_hlayout = QtWidgets.QHBoxLayout()
+        turntable_filename_hlayout.addWidget(self.txt_turntable_filename)
+        turntable_filename_hlayout.addWidget(self.btn_browse_turntable)
+        pre_form.addRow(turntable_filename_hlayout)
+
+        vlayout.addLayout(pre_form)
+        self.vlayout.addSpacing(15)
+
+    def playblast_callback(self):
```

Comment:
This has a lot of logic which looks related to the actual turntabling inside the UI logic. What do you think?

Reply:
It's mostly to keep the turntable functionality itself as pure as possible. 
The UI is for turntabling, therefore it needs to reflect what can and can be done, from a UI level, however, the functions can still be freestanding. 
Most of the logic is validity checking of user input that happens in the UI, so I feel it is not wrong to keep that logic in the UI itself.